slides/week3/week3.org

```diff
@@ -2,66 +2,158 @@
 #+AUTHOR: Kristaps Berzinch, Raymond Ortiz
 #+EMAIL: kristaps@robojackets.org, rortiz9@gatech.edu
 
-* TODO Needs speaker notes
-
 * What are we doing today?
 - Data structures and algorithms in C++
 - Branching and merging in Git
+- Using light sensor to count grey squares
+
+* What is an object?
+- a module of reusable code
+  - maintains some sort of state
+    - variables
+  - have behaviors
+    - methods
+- can have multiple instances of the same object
+
+* Arrays
+- an object that can contain a set number of other objects
+- the size of the array cannot change
+#+BEGIN_SRC C++
+  // creates an array of integers with size 3
+  #include <array>
 
-* What's a constructor?
-- A special type of function that instantiates a class
-- Basically, it sets up an object and returns it
+  array<int, 3> name = {1, 2, 3};
+#+END_SRC
 
-* What's a container?
-- An object you can use to store other objects
-- Objects are identified by keys (which can also be objects)
-- Some important properties:
-  - *Ordered:* the contents are in a specific order
-  - *Associative:* two objects are associated with each other within the container
-  - *Set:* the value of an object is its key
-  - *Unique keys:* no two objects can have the same key
+* Array methods
+| =at=    | returns the value at a specific index |
+| =front= | returns the first item                |
+| =back=  | returns the last item                 |
+| =size=  | returns the number of elements        |
 
-* Common containers
+* Vectors
 #+BEGIN_NOTES
-Note that in most languages, sets and maps are unorded. C++ also has unordered sets and maps.
+initializer lists require c++11 for vectors.
 #+END_NOTES
-| =array= | a static-length ordered list of objects |
-| =vector= | a dynamic-length ordered list of objects |
-| =set= | an ordered group of unique objects |
-| =unordered_set= | an unordered group of unique objects |
-| =map= | an ordered group of key-value pairs, with unique keys |
-| =unordered_map= | an unordered group of key-value pairs, with unique keys |
+- an object that can contain a variable number of other objects
+- you can keep adding elements to the vector
+- order of insertion in maintained
+#+BEGIN_SRC C++
+  #include <vector>
 
-* Arrays
-| =at= | returns the value at a specific index |
-| =front= | returns the first item |
-| =back= | returns the last item |
-| =size= | returns the number of elements |
+  vector<int> name = {1, 2, 3};
+#+END_SRC
 
-* Vectors
+* Vector methods
 #+BEGIN_NOTES
 Vectors can grow as items are added, whereas arrays are static.
 #+END_NOTES
-- Everything from the last slide, and...
-| =clear= | removes all items |
-| =insert= | inserts an item at a specific index |
-| =push_back= | adds a given element to the end |
-| =pop_back= | removes the last element |
+- Everything from the arrays slide, and...
```

Comment:
If you wanted, you could link to the arrays slide here via [org-store-link](http://orgmode.org/manual/Handling-links.html). `C-c l` with point on the header for the arrays slide, and insert the link via `C-c C-l`.